src/main/scala/org/ensime/core/Completion.scala

```diff
@@ -66,7 +66,7 @@ trait CompletionControl {
 
     val context = x.get match {
       case Left(tree) => {
-        logger.debug("Completing at tree:" + showRaw(tree))
+        logger.debug("Completing at tree:" + showRaw(tree).take(200))
```

Comment:
Can I suggest you add:
```scala
if(logger.debugEnabled)
  logger.debug("Completing at tree:" + showRaw(tree).take(200))
```
Which saves the string generation (I'm guessing it can be huge) if its not needed.